homeassistant/components/ozw/lock.py

```diff
@@ -74,6 +75,19 @@ def is_locked(self):
         """Return a boolean for the state of the lock."""
         return bool(self.values.primary.value)
 
+    @property
+    def changed_by(self):
+        """Last change triggered by."""
+        node: OZWNode = self.values.primary.node
+        user_code = node.get_value(
+            CommandClass.NOTIFICATION, ValueIndex.NOTIFICATION_EVENT_PARAMETER_USER_CODE
+        )
+
+        if user_code is None:
+            return
+
+        return f"{user_code.value}"
```

Comment:
Isn't the user code a credential? We don't set credentials in the state machine.

Reply:
> Isn't the user code a credential? We don't set credentials in the state machine.

No, OZW calls it the user code but it is more like the slot ID for a user. What's stored in USER_CODE is the Slot ID of the user, not the credential itself.

Reply:
Why do you convert it to a string?

Reply:
According to our dev docs this property should return a string. If the values is not a string from the beginning its good to copy it.

Whats the native value type?

https://developers.home-assistant.io/docs/core/entity/lock#properties

Reply:
https://github.com/cgarwood/python-openzwave-mqtt/blob/master/openzwavemqtt/models/value.py#L28
The return type of OZWNode.value is Any - how ever, what's sent on the MQTT topic for this value is a byte.
```
{
    "Label": "User Code",
    "Value": 0,
    "Units": "",
    "ValueSet": true,
    "ValuePolled": false,
    "ChangeVerified": false,
    "Min": 0,
    "Max": 255,
    "Type": "Byte",
    "Instance": 1,
    "CommandClass": "COMMAND_CLASS_NOTIFICATION",
    "Index": 260,
    "Node": 49,
    "Genre": "User",
    "Help": "User Code ID that was used",
    "ValueIDKey": 73183494772899857,
    "ReadOnly": true,
    "WriteOnly": false,
    "Event": "valueChanged",
    "TimeStamp": 1611850845
}
```

So I think this is the proper way, maybe not the most pythonic way though.

Reply:
Ok, then let's keep the string formatting.